### PR TITLE
node, ir: rename `morph` config to `fschain`

### LINKED ISSUE #3021: Rename morph config to fschain
## Is your feature request related to a problem? Please describe.
I'm always frustrated when people ask me what "morph" means in the node configuration. We need to discontinue this term, all of the current documentation uses "FS chain" and neo-exporter uses "fschain" for a similar configuration.

## Describe the solution you'd like
Rename `morph` to `fschain`. With a transition period for a single release where we would accept both (and warn about `morph` use). Functionally it stays the same with the same inner options.

## Additional context
#2702
https://github.com/nspcc-dev/neofs-spec/pull/99
https://github.com/nspcc-dev/neofs-api/pull/310
https://github.com/nspcc-dev/neofs-contract/pull/447

### PR DESCRIPTION
**Pull request by End-rey**
Closes #3021.

Not sure about 
https://github.com/nspcc-dev/neofs-node/blob/3f6d545d29dcaa49a0de9264ffd4717d2bbdc5f9/pkg/innerring/state.go#L22
and 
https://github.com/nspcc-dev/neofs-node/blob/3f6d545d29dcaa49a0de9264ffd4717d2bbdc5f9/cmd/neofs-node/config.go#L525

### CODE REVIEW

**Comment by roman-khimov** [comment]
> Not sure about

These are DB keys, not a big problem.

File: cmd/neofs-node/config.go [hunk 1]
```diff
@@ -522,7 +522,7 @@ type cfgReputation struct {
 	localTrustCtrl *trustcontroller.Controller
 }
 
-var persistateSideChainLastBlockKey = []byte("side_chain_last_processed_block")
+var persistateFSChainLastBlockKey = []byte("fs_chain_last_processed_block")
```
Comment (roman-khimov) [inline cmd/neofs-node/config.go:525]: This is an incompatible change, we either need to keep the value or provide migration.

File: pkg/innerring/state.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ const voteMethod = "vote"
 
 var (
 	persistateMainChainLastBlockKey = []byte("main_chain_last_processed_block")
-	persistateSideChainLastBlockKey = []byte("side_chain_last_processed_block")
+	persistateFSChainLastBlockKey   = []byte("fs_chain_last_processed_block")
```
Comment (roman-khimov) [inline pkg/innerring/state.go:22]: This one too.

File: cmd/neofs-ir/defaults.go [hunk 3]
```diff
@@ -109,9 +109,7 @@ func defaultConfiguration(cfg *viper.Viper) {
 	cfg.SetDefault("indexer.cache_timeout", 15*time.Second)
 
 	// extra fee values for working mode without notary contract
-	cfg.SetDefault("fee.main_chain", 5000_0000)                  // 0.5 Fixed8
-	cfg.SetDefault("fee.side_chain", 2_0000_0000)                // 2.0 Fixed8
-	cfg.SetDefault("fee.named_container_register", 25_0000_0000) // 25.0 Fixed8
```
Comment (roman-khimov) [inline cmd/neofs-ir/defaults.go:114]: This one is (incorrectly) mentioned in `pkg/innerring/config/fee.go`, can be adjusted as well.

File: pkg/innerring/config.go [hunk 4]
```diff
@@ -20,8 +20,8 @@ import (
 )
 
 // Various configuration paths.
-const (
-	cfgPathFSChain               = "morph"
+var (
```
Comment (roman-khimov) [inline pkg/innerring/config.go:23]: Shouldn't be needed.

File: pkg/innerring/innerring.go [hunk 5]
```diff
@@ -327,10 +328,21 @@ func New(ctx context.Context, log *zap.Logger, cfg *viper.Viper, errChan chan<-
...
 	}
 
-	morphChain := chainParams{
+	if cfg.IsSet(morphPrefix+".endpoints") || cfg.IsSet(morphPrefix+".consensus") {
+		log.Warn("section name 'morph' is deprecated, use 'fschain'")
+		cfgPathFSChain = morphPrefix
+	}
+	if cfg.IsSet(fsChainPrefix+".endpoints") || cfg.IsSet(fsChainPrefix+".consensus") {
+		cfgPathFSChain = fsChainPrefix
+		log.Info("use 'fschain' prefix", zap.String("prefix", cfgPathFSChain))
```
Comment (roman-khimov) [inline pkg/innerring/innerring.go:337]: Not worth logging.

File: pkg/innerring/innerring.go [hunk 6]
```diff
@@ -327,10 +328,21 @@ func New(ctx context.Context, log *zap.Logger, cfg *viper.Viper, errChan chan<-
 		log.Warn("can't get last processed FS chain block number", zap.Error(err))
 	}
 
-	morphChain := chainParams{
+	if cfg.IsSet(morphPrefix+".endpoints") || cfg.IsSet(morphPrefix+".consensus") {
+		log.Warn("section name 'morph' is deprecated, use 'fschain'")
```
Comment (roman-khimov) [inline pkg/innerring/innerring.go:332]: config section 'morph' is deprecated...

File: CHANGELOG.md [hunk 7]
```diff
@@ -49,6 +49,7 @@ attribute, which is used for container domain name in NNS contracts (#2954)
 - Log sampling is disabled by default now (#3011)
 - EACL is no longer considered for system role (#2972)
 - Deprecate peapod substorage (#3013)
+- Rename in node and inner ring configs `morph` section to `fschain` (#3028)
```
Comment (roman-khimov) [inline CHANGELOG.md:52]: `morph` configuration section renamed to `fschain` both in IR and SN

File: CHANGELOG.md [hunk 8]
```diff
@@ -72,6 +73,10 @@ introduced in version 0.22.3 and support for binary keys was removed from
 other components in 0.33.0 and 0.37.0. Please migrate to wallets (see 0.37.0
 notes) if you've not done it previously.
 
+The section `morph` in the config has been renamed to `fschain`. This is now
```
Comment (roman-khimov) [inline CHANGELOG.md:76]: ... This version still supports the old section name, but this compatibility code will be removed in the next release. Please rename `morph` to `fschain` in your configuration files.

File: cmd/neofs-node/config/fschain/config.go [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+import (
+	"fmt"
+	"time"
+
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-node/config"
+)
+
+const (
+	subsection      = "fschain"
+	morphSubsection = "morph"
```
Comment (carpawell) [inline cmd/neofs-node/config/fschain/config.go:12]: can you add "deprecated" somehow to the var name?

File: pkg/innerring/config.go (hunk 4 above)
Comment (carpawell) [reply pkg/innerring/config.go:23]: you did it many times below, what made you do so?

File: cmd/neofs-adm/internal/modules/morph/initialize.go [hunk 10]
```diff
@@ -43,7 +43,7 @@ type initializeContext struct {
 	ContractPath string
 }
 
-func initializeSideChainCmd(cmd *cobra.Command, _ []string) error {
+func initializeFSChainCmd(cmd *cobra.Command, _ []string) error {
```
Comment (carpawell) [inline cmd/neofs-adm/internal/modules/morph/initialize.go:46]: the dir itself may also be renamed

**Review by carpawell — COMMENTED** [review]
Shouldn't we drop `cmd/neofs-node/config/morph`?
My vote is for migration for all the db keys that have `morph`.

File: pkg/innerring/config.go (hunk 4 above)
Comment (End-rey) [reply pkg/innerring/config.go:23]: I did this to change the name depending on which one was provided in the config.

File: pkg/innerring/innerring.go [hunk 11]
```diff
@@ -321,16 +322,41 @@ func New(ctx context.Context, log *zap.Logger, cfg *viper.Viper, errChan chan<-
...
 	if err != nil {
 		fromFSChainBlock = 0
 		log.Warn("can't get last processed FS chain block number", zap.Error(err))
 	}
 
-	morphChain := chainParams{
+	// migration for deprecated DB key
+	if fromFSChainBlock == 0 && fromDeprectedSidechanBlock != fromFSChainBlock {
+		fromFSChainBlock = fromDeprectedSidechanBlock
+		err = server.persistate.SetUInt32(persistateFSChainLastBlockKey, fromFSChainBlock)
```
Comment (carpawell) [inline pkg/innerring/innerring.go:338]: should we drop the previous value? @roman-khimov

**Review by roman-khimov — APPROVED** [review]

**Comment by roman-khimov** [comment]
Tests need to be fixed.

> output: Error: unknown command "morph" for "neofs-adm"

We need to keep compatibility here as well.

**Review by carpawell — APPROVED** [review]

**Pull request merged**